src/transformers/models/ict/modeling_ict.py

```diff
@@ -0,0 +1,837 @@
+
+        return outputs
+
+
+ICT_START_DOCSTRING = r"""
+    This model inherits from [`PreTrainedModel`]. Check the superclass documentation for the generic methods the
+    library implements for all its model (such as downloading or saving, resizing the input embeddings, pruning heads
+    etc.)
+
+    This model is a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass. Use it
+    as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`IctConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+ICT_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, height * width)`):
+            Pixel values. Pixel values can be obtained using [`AutoImageProcessor`]. See [`IctImageProcessor.__call__`]
+            for details.
+        clusters (`np.ndarray`, of shape `(n_clusters, 3)`):
+            Clusters used to quantize the image of shape `(n_clusters, 3)` before being fed to Guided Upsampler.
+        bool_masked_pos (`torch.BoolTensor` of shape `(batch_size, height * width)`, *optional*):
+            Boolean masked positions. Indicates which patches are masked (1) and which aren't (0). Generate random
+            masks if not provided.
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+@add_start_docstrings(ICT_START_DOCSTRING)
+class IctModel(IctPretrainedModel):
+    config_class = IctConfig
+
+    def __init__(self, config: IctConfig, use_mask_token: bool = True):
+        super().__init__(config)
+
+        self.config = config
+        self.transformer = IctTransformerModel(config, use_mask_token=use_mask_token)
+        self.guided_upsampler = IctGuidedUpsampler(config)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.transformer.embeddings.token_embedding
+
+    @add_start_docstrings_to_model_forward(ICT_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=MaskedImageModelingOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        pixel_values: Optional[torch.Tensor],
```

Comment:
Just a general question, is there a reason we allow `Optional` for this param? Seems like this is pretty necessary?